packages/client/lib/functions/dpopUtil.ts

```diff
@@ -0,0 +1,33 @@
+import { dpopResourceAuthenticateError, dpopTokenRequestNonceError } from '@sphereon/oid4vc-common';
+import { OpenIDResponse } from 'oid4vci-common';
+
+export function dPoPShouldRetryRequestWithNonce(response: OpenIDResponse<unknown, unknown>) {
+  if (response.errorBody && response.errorBody.error === dpopTokenRequestNonceError) {
+    const dPoPNonce = response.errorBody.headers.get('DPoP-Nonce');
+    if (!dPoPNonce) {
+      throw new Error('The DPoP nonce was not returned');
+    }
+
+    return { ok: true, dpopNonce: dPoPNonce } as const;
```

Comment:
Can we create a small type for this? The ok is not immediately clear. I think having the http  status code instead of ok would be nicer. The 401 is because we handle it nicely I guess, but it could also happen with a 400. So probably just using the http status is quite useful

Reply:
I created a type. Hmmmmm, Not sure about the HTTP status return. Because then, at a higher level, you need to know which status codes trigger the retry.

The 401 is required for Resource Requests. https://www.rfc-editor.org/rfc/rfc9449.html#name-resource-server-provided-no